src/main/java/io/github/moulberry/notenoughupdates/overlays/BazaarSearchOverlay.java

```diff
@@ -0,0 +1,574 @@
+		if (scaledResolution.getScaleFactor() >= 4) {
+			topY = height / 2 - h / 2 + 5;
+		}
+
+		if (!Mouse.getEventButtonState() && Mouse.getEventButton() == -1 && searchFieldClicked) {
+			textField.mouseClickMove(mouseX - 2, topY + 10, 0, 0);
+		}
+
+		if (Mouse.getEventButton() != -1) {
+			searchFieldClicked = false;
+		}
+
+		if (Mouse.getEventButtonState()) {
+			if (mouseY > topY && mouseY < topY + 20) {
+				if (mouseX > width / 2 - 100) {
+					if (mouseX < width / 2 + 49) {
+						searchFieldClicked = true;
+						textField.mouseClicked(mouseX - 2, mouseY, Mouse.getEventButton());
+
+						if (Mouse.getEventButton() == 1) {
+							searchString = "";
+							synchronized (autocompletedItems) {
+								autocompletedItems.clear();
+							}
+						}
+					} else if (mouseX < width / 2 + 75) {
+						searchStringExtra = "";
+						close();
+					} else if (mouseX < width / 2 + 100) {
+						searchStringExtra = "";
+						close();
+						Minecraft.getMinecraft().thePlayer.sendQueue.addToSendQueue(new C0DPacketCloseWindow(Minecraft.getMinecraft().thePlayer.openContainer.windowId));
+						NotEnoughUpdates.INSTANCE.openGui = new GuiScreenElementWrapper(new NEUConfigEditor(
+							NotEnoughUpdates.INSTANCE.config, "Bazaar Search GUI"));
+					}
+				}
+			} else if (Mouse.getEventButton() == 0) {
+				int num = 0;
+				synchronized (autocompletedItems) {
+					for (String str : autocompletedItems) {
+						JsonObject obj = NotEnoughUpdates.INSTANCE.manager.getItemInformation().get(str);
+						if (obj != null) {
+							ItemStack stack = NotEnoughUpdates.INSTANCE.manager.jsonToStack(obj);
+							if (mouseX >= width / 2 - 96 && mouseX <= width / 2 + 96 && mouseY >= topY + 30 + num * 22 &&
+								mouseY <= topY + 30 + num * 22 + 20) {
+								searchString = Utils.cleanColour(stack.getDisplayName().replaceAll("\\[.+]", "")).trim();
+								if (searchString.contains("Enchanted Book") && str.contains(";")) {
+									String[] lore = NotEnoughUpdates.INSTANCE.manager.getLoreFromNBT(stack.getTagCompound());
+									String[] split = Utils.cleanColour(lore[0]).trim().split(" ");
+									split[split.length - 1] = "";
+
+									searchString = StringUtils.join(split, " ").trim();
+								}
+
+								searchStringExtra = " ";
+
+								close();
+								return;
+							}
+
+							if (++num >= 5) break;
+						}
+					}
+				}
+
+				if (NotEnoughUpdates.INSTANCE.config.bazaarTweaks.showPastSearches) {
+					for (int i = 0; i < 5; i++) {
+						if (i >= NotEnoughUpdates.INSTANCE.config.hidden.previousBazaarSearches.size()) break;
+
+						String s = NotEnoughUpdates.INSTANCE.config.hidden.previousBazaarSearches.get(i);
+						if (mouseX >= width / 2 - 95 && mouseX <= width / 2 + 95 &&
+							mouseY >= topY + 45 + AUTOCOMPLETE_HEIGHT + i * 10 &&
+							mouseY <= topY + 45 + AUTOCOMPLETE_HEIGHT + i * 10 + 10) {
+							searchString = s;
+							searchStringExtra = "";
+							close();
+							return;
+						}
+					}
+				}
+			}
+		}
+	}
+}
```

Comment:
Looking at the AH code duplicated has made me sick, so i can understand why you dont want to bother generalizing the AH overlay.

Reply:
I don't think the extra effort generalizing this kind of search GUI is needed for anything else except bz and ah atm. It took me under 10 minutes to cut the star stuff from ah and renamed all endpoints to bazaar. When Hypixel adds new stuff where we could use the same GUI again, I will generalize it then.